# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/brush_blend.glsl
# Commit: 88039c2ec513
# Full Hash: 88039c2ec5138a0185cfdd5260adefe078e55a0f
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-04-06 09:21:15
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1825518 - Use highp for v_color_mat varying to avoid crash on Mali-T devices. r=gfx-reviewers,lsalzman
#   
#   Since landing bug 1823411 we have been receiving crash reports on a
#   variety of Mali-T devices when attempting to compile the brush_blend
#   shader. This appears to be due to changing v_color_mat to mediump,
# ==============================================================================

diff -r bba23a9ea29c -r 88039c2ec513 gfx/wr/webrender/res/brush_blend.glsl
--- a/gfx/wr/webrender/res/brush_blend.glsl	Wed Apr 05 15:52:36 2023 +0000
+++ b/gfx/wr/webrender/res/brush_blend.glsl	Wed Apr 05 16:17:19 2023 +0000
@@ -30,7 +30,9 @@
 #define v_op v_op_table_address_vec.x
 #define v_table_address v_op_table_address_vec.y
 
-flat varying mediump mat4 v_color_mat;
+// We must keep this as highp as we encoutered shader compilation crashes on
+// Mali-T devices when mediump.
+flat varying highp mat4 v_color_mat;
 // The function to use for each component of a component transfer filter. Using a int[4]
 // or ivec4 (with each element or component containing the function for each component) has
 // ran in to bugs 1695912 and 1731758, so instead use a vec4 and cast the values to/from floats.
